prefq/examples/updated_imitation_modules.py

```diff
@@ -0,0 +1,232 @@
+            accepting two arguments, the Env to be wrapped and the environment index,
+            and returning the wrapped Env.
+        env_make_kwargs: The kwargs passed to `spec.make`.
+
+    Returns:
+        A VecEnv initialized with `n_envs` environments.
+    """
+    # Resolve the spec outside of the subprocess first, so that it is available to
+    # subprocesses running `make_env` via automatic pickling.
+    # Just to ensure packages are imported and spec is properly resolved
+    tmp_env = gymnasium.make(env_name, render_mode=render_mode)
+    tmp_env.close()
+    spec = tmp_env.spec
+    env_make_kwargs = env_make_kwargs or {}
+
+    def make_env(i: int, this_seed: int) -> gymnasium.Env:
+        # Previously, we directly called `gym.make(env_name)`, but running
+        # `imitation.scripts.train_adversarial` within `imitation.scripts.parallel`
+        # created a weird interaction between Gym and Ray -- `gym.make` would fail
+        # inside this function for any of our custom environment unless those
+        # environments were also `gym.register()`ed inside `make_env`. Even
+        # registering the custom environment in the scope of `make_vec_env` didn't
+        # work. For more discussion and hypotheses on this issue see PR #160:
+        # https://github.com/HumanCompatibleAI/imitation/pull/160.
+        assert env_make_kwargs is not None  # Note: to satisfy mypy
+        assert spec is not None  # Note: to satisfy mypy
+        env = gymnasium.make(
+            spec,
+            max_episode_steps=max_episode_steps,
+            render_mode=render_mode,
+            **env_make_kwargs,
+        )
+
+        # Seed each environment with a different, non-sequential seed for diversity
+        # (even if caller is passing us sequentially-assigned base seeds). int() is
+        # necessary to work around gym bug where it chokes on numpy int64s.
+        env.reset(seed=int(this_seed))
+        # NOTE: we do it here rather than on the final VecEnv, because
+        # that would set the same seed for all the environments.
+
+        # Use Monitor to record statistics needed for Baselines algorithms logging
+        # Optionally, save to disk
+        log_path = None
+        if log_dir is not None:
+            log_subdir = os.path.join(log_dir, "monitor")
+            os.makedirs(log_subdir, exist_ok=True)
+            log_path = os.path.join(log_subdir, f"mon{i:03d}")
+
+        env = monitor.Monitor(env, log_path)
+
+        if post_wrappers:
+            for wrapper in post_wrappers:
+                env = wrapper(env, i)
+
+        return env
+
+    env_seeds = make_seeds(rng, n_envs)
+    env_fns: List[Callable[[], gymnasium.Env]] = [
+        functools.partial(make_env, i, s) for i, s in enumerate(env_seeds)
+    ]
+    if parallel:
+        # See GH hill-a/stable-baselines issue #217
+        return SubprocVecEnv(env_fns, start_method="forkserver")
+    else:
+        return DummyVecEnv(env_fns)
+
+
+@overload
```

Comment:
Why do we need these?

Reply:
The make_vec_env() function calls the make_seeds() function - I do not fully understand how they work, or what they do, however, within the imitation package these functions were defined like this just below make_vec_env(), so I just copied them in order to ensure the same functionality under the updated version

Reply:
by simply using "env_make_kwargs={...}" & replacing the pinned reference to imitation in "pyproject.toml" it was possible to get rid of this workaround